akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -30,6 +30,10 @@ final class Sink[-In, +Mat](private[stream] override val module: Module)
 
   override val shape: SinkShape[In] = module.shape.asInstanceOf[SinkShape[In]]
 
+  def contramap[In2](f: In2 ⇒ In): Sink[In2, Mat] = Flow.fromFunction(f).toMat(this) { (_, m) ⇒ m }
```

Comment:
@drewhk how would the implementation of  `contramapMat` differ from the actual one? as we are combining a pure function, I can't see why one would like to give something different than `Keep.right` in this case, am I missing something?